lvmd/vgservice_test.go

```diff
@@ -47,89 +50,170 @@ func (s *mockWatchServer) RecvMsg(m interface{}) error {
+				t.Fatal("not received")
+			}
+
+			select {
+			case <-ch1:
+				t.Fatal("unexpected event")
+			default:
+			}
+
+			ch2 := make(chan struct{})
+			server2 := &mockWatchServer{
+				ctx: ctx,
+				ch:  ch2,
+			}
+			go func() {
+				vgService.Watch(&proto.Empty{}, server2)
+			}()
+
+			notifier()
+
+			select {
+			case <-ch1:
+			case <-time.After(waitDuration * time.Second):
+				t.Fatal("not received")
+			}
+			select {
+			case <-ch2:
+			case <-time.After(waitDuration * time.Second):
+				t.Fatal("not received")
+			}
+
+			cancel()
+
+			select {
+			case <-done:
+			case <-time.After(waitDuration * time.Second):
+				t.Fatal("not done")
+			}
+
+		})
 	}
+}
 
-	cancel()
+func testVGService(t *testing.T, vg *command.VolumeGroup) {
 
-	select {
-	case <-done:
-	case <-time.After(1 * time.Second):
-		t.Fatal("not done")
+	// thinpool details
+	overprovisionRatio := float64(10.0)
+	poolName := "test_pool"
+	poolSize := uint64(1 << 30)
+	// TODO (leelavg): should the createpool method be moved to lvmd/test_utils.go?
+	pool, err := vg.CreatePool(poolName, poolSize)
+	if err != nil {
+		t.Fatal(err)
 	}
-}
 
-func testVGService(t *testing.T, vg *command.VolumeGroup) {
 	spareGB := uint64(1)
-	vgService, _ := NewVGService(NewDeviceClassManager([]*DeviceClass{{Name: vg.Name(), VolumeGroup: vg.Name(), SpareGB: &spareGB}}))
-	res, err := vgService.GetLVList(context.Background(), &proto.GetLVListRequest{DeviceClass: vg.Name()})
+	thickdev := vg.Name()
+	thindev := poolName
+	vgService, _ := NewVGService(
+		NewDeviceClassManager(
+			[]*DeviceClass{
+				{
+					// volumegroup target
+					Name:        thickdev,
+					VolumeGroup: vg.Name(),
+					SpareGB:     &spareGB,
+				},
+				{
+					// thinpool target
+					Name:        thindev,
+					VolumeGroup: vg.Name(),
+					SpareGB:     &spareGB,
+					Type:        TypeThin,
+					ThinPoolConfig: &ThinPoolConfig{
+						Name:               poolName,
+						OverprovisionRatio: overprovisionRatio,
+					},
+				},
+			},
+		),
+	)
+
+	// thick lvs
+	res, err := vgService.GetLVList(context.Background(), &proto.GetLVListRequest{DeviceClass: thickdev})
 	if err != nil {
 		t.Fatal(err)
 	}
 	numVols1 := len(res.GetVolumes())
 	if numVols1 != 0 {
 		t.Errorf("numVolumes must be 0: %d", numVols1)
 	}
+
+	// thin lvs
+	res, err = vgService.GetLVList(context.Background(), &proto.GetLVListRequest{DeviceClass: thickdev})
+	if err != nil {
+		t.Fatal(err)
+	}
+	numVols1 = len(res.GetVolumes())
+
+	if numVols1 != 0 {
+		t.Errorf("numVolumes must be 0: %d in pool %s", numVols1, poolName)
+	}
+
+	// create thick volume
 	testtag := "testtag"
 	_, err = vg.CreateVolume("test1", 1<<30, []string{testtag}, 0, "", nil)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	res, err = vgService.GetLVList(context.Background(), &proto.GetLVListRequest{DeviceClass: vg.Name()})
+	// thick lv validation
+	res, err = vgService.GetLVList(context.Background(), &proto.GetLVListRequest{DeviceClass: thickdev})
```

Comment:
The tests for VG are replaced for thin-pool and tests for VG are looks removed. Would you keep the tests for VG? It is similar to [this comment](https://github.com/topolvm/topolvm/pull/448#discussion_r839364202).

Reply:
This comment is my mistake. `thickdev` is not `thindev`.

Reply:
- yes, as mentioned no existing unit test is removed.